src/pages/CountryDetails/CountryDetails.tsx

```diff
@@ -0,0 +1,110 @@
+import { useParams, Link, useNavigate } from 'react-router-dom';
+import { useFetch } from '../../pages/Countries/hooks/useFetch';
+import { CountryDetails } from '@types/types';
+import { Toast } from '@/components/Toast/Toast';
+import { LoadingOverlay } from '@/components/LoadingOverlay/LoadingOverlay';
+
+import './CountryDetails.scss';
+
+export function CountryDetail() {
+  const { countryCode } = useParams();
+  const navigate = useNavigate();
+
+  const {
+    data: country,
+    loading,
+    error,
+  } = useFetch<CountryDetails>(
+    `https://restcountries.com/v3.1/alpha/${countryCode}?fields=name,population,region,subregion,capital,flags,cca3,borders,currencies,languages,tld`
+  );
+
+  return (
+    <>
+      {
+        <LoadingOverlay
+          isLoading={loading}
+          message="Loading country details..."
+        />
+      }
+
+      {error && (
+        <Toast
+          message={`Error: ${error.message}`}
+          type="error"
+          duration={null}
+        />
+      )}
+      {!country && !loading && (
+        <Toast
+          message="no country found"
+          type="warning"
+          duration={null}
+        />
+      )}
+
+      <article className="country-details">
```

Comment:
zależy gdzie masz znacznik 'main' ustawiony, ale jeżeli nie ma go na tej stronie nigdzie to tutaj zamiast article powinien być - jest to główna zawartość tej strony